platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/OutputWire.java

```diff
@@ -0,0 +1,236 @@
+     * @return this
+     */
+    @NonNull
+    public final OutputWire<OUT> solderTo(@NonNull final InputWire<OUT, ?> inputWire) {
+        return solderTo(inputWire, false);
+    }
+
+    /**
+     * Specify an input wire where output data should be passed. This forwarding operation respects back pressure.
+     *
+     * <p>
+     * Soldering is the act of connecting two wires together, usually by melting a metal alloy between them. See
+     * <a href="https://en.wikipedia.org/wiki/Soldering">wikipedia's entry on soldering</a>.
+     *
+     * <p>
+     * Forwarding should be fully configured prior to data being inserted into the system. Adding forwarding
+     * destinations after data has been inserted into the system is not thread safe and has undefined behavior.
+     *
+     * @param inputWire the input wire to forward output data to
+     * @param inject    if true, then the output data will be injected into the input wire, ignoring back pressure
+     * @return this
+     */
+    @NonNull
+    public final OutputWire<OUT> solderTo(@NonNull final InputWire<OUT, ?> inputWire, final boolean inject) {
+        model.registerEdge(name, inputWire.getTaskSchedulerName(), inputWire.getName(), inject);
+        if (inject) {
+            forwardingDestinations.add(inputWire::inject);
+        } else {
+            forwardingDestinations.add(inputWire::put);
+        }
+        return this;
+    }
+
+    /**
+     * Specify a consumer where output data should be forwarded.
+     *
+     * <p>
+     * Soldering is the act of connecting two wires together, usually by melting a metal alloy between them. See
+     * <a href="https://en.wikipedia.org/wiki/Soldering">wikipedia's entry on soldering</a>.
+     *
+     * <p>
+     * Forwarding should be fully configured prior to data being inserted into the system. Adding forwarding
+     * destinations after data has been inserted into the system is not thread safe and has undefined behavior.
+     *
+     * @param handlerName the name of the consumer
+     * @param handler     the consumer to forward output data to
+     * @return this
+     */
+    @NonNull
+    public OutputWire<OUT> solderTo(@NonNull final String handlerName, @NonNull final Consumer<OUT> handler) {
+        model.registerEdge(name, handlerName, "", false);
+        forwardingDestinations.add(Objects.requireNonNull(handler));
+        return this;
+    }
+
+    /**
+     * Build a {@link WireFilter} that is soldered to the output of this wire.
+     *
+     * @param name      the name of the filter
+     * @param predicate the predicate that filters the output of this wire
+     * @return the filter
+     */
+    @NonNull
+    public OutputWire<OUT> buildFilter(@NonNull final String name, @NonNull final Predicate<OUT> predicate) {
+        final WireFilter<OUT> filter =
+                new WireFilter<>(model, Objects.requireNonNull(name), Objects.requireNonNull(predicate));
+        solderTo(filter.getName(), filter);
```

Comment:
It may be more clear to users if this method did not exist on its own, but was a variant of `solderTo`, such as `filterAndSolderTo`

Reply:
Another option is to only support soldering to a single output wire, and make it up to the caller to combine output wires (filters, splitters, transformers) into a single output wire.

Reply:
I don't understand your first comment. I'm not sure how the ordering matters. We should discuss.

> Another option is to only support soldering to a single output wire or a list of wires, and make it up to the caller to combine output wires

The reason for this is unclear to me. When writing tests, it was quite convenient to allow the framework to manage this for me. Let's also discuss this.

Reply:
I think the confusing part is that the methods in this class return `this` sometimes, and sometimes they return  `filter` or `splitter`. I feel like `solderTo` should return `void` to avoid this confusion.

Reply:
> I think the confusing part is that the methods in this class return this sometimes, and sometimes they return filter or splitter. I feel like solderTo should return void to avoid this confusion.

Makes sense, change made. `solderTo()` is now a void method.

Reply:
Per my discussion with @Cody, it would also be helpful to document that the `builderFilter` and similar methods for transformer and splitter actually return the output wire of the fitler/transformer/splitter which is a newly created output wire.

Reply:
added requested documentation